daemons/controld/controld_fencing.c

```diff
@@ -478,55 +498,51 @@ tengine_stonith_notify(stonith_t *st, stonith_event_t *st_event)
          * our subsequent election votes as "not part of our cluster".
          */
         crm_crit("We were allegedly just fenced by %s for %s!",
-                 st_event->executioner? st_event->executioner : "the cluster",
-                 st_event->origin); /* Dumps blackbox if enabled */
+                 executioner, event->origin); // Dumps blackbox if enabled
         if (fence_reaction_panic) {
             pcmk__panic(__func__);
         } else {
             crm_exit(CRM_EX_FATAL);
         }
-        return;
+        return; // Should never get here
     }
 
-    /* Update the count of stonith failures for this target, in case we become
+    /* Update the count of fencing failures for this target, in case we become
      * DC later. The current DC has already updated its fail count in
      * tengine_stonith_callback().
      */
-    if (!AM_I_DC && pcmk__str_eq(st_event->operation, T_STONITH_NOTIFY_FENCE, pcmk__str_casei)) {
-        if (st_event->result == pcmk_ok) {
-            st_fail_count_reset(st_event->target);
+    if (!AM_I_DC
+        && pcmk__str_eq(event->operation, T_STONITH_NOTIFY_FENCE,
+                        pcmk__str_casei)) {
```

Comment:
shouldn't we compare the operation case-sensitive?

Reply:
Yes, a lot of the case-insensitive comparisons are left over from when we converted from safe_str_eq() (which was always case-insensitive and used where it really shouldn't have been) to pcmk__str_eq().

Reply:
Rebased on current main (no code changes) and added commit on top for this

Reply:
I know where it is coming from. Just mention it when I stumble over it.